examples/llava/mtmd.cpp

```diff
@@ -123,14 +130,14 @@ mtmd_input_chunks * mtmd_tokenize(mtmd_context * ctx,
             std::move(tokens),
             {},
         };
-        output->emplace_back(std::move(chunk));
+        output.emplace_back(std::move(chunk));
 
         if (&parts.back() != &part) {
```

Comment:
Not 100% sure, but I think this logic does not handle the case where the text ends with an image marker:

```
<some_text><image_marker>
```

For Gemma3 this will not happen because wrap the image marker with text on both sides, but maybe for other models it could happen? If it cannot happen for sure, then this `if` should become assert.

Reply:
The current logic will produce an empty text chunk in the end if the image marker is placed at the end on the input prompt. This is an effect from `string_split_str`

For example, this code:

```cpp
auto test = string_split_str("123aa456aa", "aa");
for (auto & p : test) printf("'%s'\n", p.c_str());
```

Will output:

```
'123'
'456'
''
```

I think having an empty chunk in the end is expected for now, but I should document it better.

If we don't want this empty chunk, the proper way is to stop using `string_split_str` and to write our own code to do string matching / splitting.

In reality, this will almost never happen because user always input a prompt with a generation prefix, something like `<user>\nwhat do you see?<image></user><assistant>`

Reply:
Sorry I missed one line of code:

```cpp
auto tokens = mtmd_tokenize_text_internal(...);
if (tokens.empty()) {
    continue;
}
```

So that means there is no empty chunk being added, the case where image marker placed in the end is correctly handled.

(This piece of code was firstly introduced from my first attempt to refactor vision API, so yeah it's quite hacky)